Crypto/PubKey/ECC/Prim.hs

```diff
@@ -30,9 +31,9 @@ scalarGenerate curve = generateBetween 1 (n - 1)
 -- | Elliptic Curve point negation:
 -- @pointNegate c p@ returns point @q@ such that @pointAdd c p q == PointO@.
 pointNegate :: Curve -> Point -> Point
-pointNegate _           PointO     = PointO
-pointNegate CurveFP{}  (Point x y) = Point x (-y)
-pointNegate CurveF2m{} (Point x y) = Point x (x `addF2m` y)
+pointNegate _            PointO     = PointO
+pointNegate (CurveFP c) (Point x y) = Point x (ecc_p c - y)
```

Comment:
that looks like it's fixing a serious bug in pointNegate, or changing the sign of y eventually "even out" later in the modular operation ?

Reply:
I found the bug when writing P256 tests and comparing results with ECC.
But so far this has not been serious: `pointNegate` is not exposed, only used internally in a way that does not matter.